src/App/pages/MainPage/components/Filter/components/MultiDropdown/MultiDropdown.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import Input from 'components/Input';
+import ArrowDownIcon from 'components/Icons/ArrowDownIcon';
+import styles from './MultiDropDown.module.scss';
+
+export type Option = {
+  /** Ключ варианта, используется для отправки на бек/использования в коде */
+  key: string;
+  /** Значение варианта, отображается пользователю */
+  value: string;
+};
+
+/** Пропсы, которые принимает компонент Dropdown */
+export type MultiDropdownProps = {
+  className?: string;
+  /** Массив возможных вариантов для выбора */
+  options: Option[];
+  /** Текущие выбранные значения поля, может быть пустым */
+  value: Option[];
+  /** Callback, вызываемый при выборе варианта */
+  onChange: (value: Option[]) => void;
+  /** Заблокирован ли дропдаун */
+  disabled?: boolean;
+  /** Возвращает строку которая будет выводится в инпуте. В случае если опции не выбраны, строка должна отображаться как placeholder. */
+  getTitle: (value: Option[]) => string;
+};
+
+const MultiDropdown: React.FC<MultiDropdownProps> = (props) => {
+  const { className, options, value, onChange, disabled, getTitle } = props;
+
+  const [isOpened, setIsOpened] = useState(false);
+  const [search, setSearch] = useState('');
+
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const toggleDropdown = () => {
+    setIsOpened((prev) => !prev);
+  };
+
+  useEffect(() => {
+    const handleClick = (event: MouseEvent) => {
+      if (!containerRef.current) return;
+      if (!(event.target instanceof Element)) return;
+
+      const isOpen = containerRef.current.contains(event.target);
+      if (!isOpen) {
+        setIsOpened(false);
+      }
+    };
+
+    window.document.addEventListener('click', handleClick);
+    return () => {
+      window.document.removeEventListener('click', handleClick);
+    };
+  }, [setIsOpened, value]);
+
+  useEffect(() => {
+    if (isOpened) {
+      setSearch('');
+    } else {
+      setSearch(value.length ? getTitle(value) : '');
+    }
+  }, [isOpened, value, getTitle, setSearch]);
+
+  const filteredOptions = options.filter((option) => {
+    if (search === null) return true;
+    return option.value.toLocaleLowerCase().startsWith(search.toLocaleLowerCase());
+  });
+
+  return (
+    <div ref={containerRef} className={classNames(styles.container, className)}>
+      <div className={styles.inputWrapper} onClick={toggleDropdown}>
+        <Input
+          className={classNames(
+            {
+              [styles.input]: isOpened,
+            },
+            { [styles.input]: value.length === 0 },
+          )}
```

Comment:
можно сделать покороче:
```suggestion
          className={classNames((isOpened || value.length === 0) && styles.input)}
```